### PR TITLE
fix: [CDS-86876]: Adds support to show complete error message for approval change window field when create metadata api fails

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

This PR adds support to show the detailed error message when create metadata api fails for  service now approval change window field.
This would help the users to understand the error in a better way so that they can take required action if it is due to configuration miss etc - like permission not available and so on.

#### Screenshots

Before fix -->
Screenshot 2023-12-27 at 7 08 06 PM

After fix --> 

Screenshot 2023-12-27 at 7 04 42 PM

Loading behaviour (sanity)-->

Screenshot 2023-12-27 at 7 21 13 PM

General Sanity --> 

https://github.com/harness/harness-core-ui/assets/96409598/7f706543-43a2-43ae-a16d-fa6fd6ed68b0

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
trigger messagecheck

**Review by litvinovstas — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalChangeWindow.tsx [hunk 1]
```diff
@@ -87,18 +92,26 @@ export function ServiceNowApprovalChangeWindow({
...
+            {...commonMultiTypeInputProps(getString('pipeline.serviceNowApprovalStep.windowEnd'))}
+          />
+        </Layout.Horizontal>
+        {createMetadataError && (
+          <Layout.Horizontal spacing="small">
+            <Icon name="error" intent={Intent.DANGER} />
+            <Text intent={Intent.DANGER}>{getRBACErrorMessage(createMetadataError)}</Text>
+          </Layout.Horizontal>
+        )}
+      </Layout.Vertical>
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalChangeWindow.tsx:114]: Do we need test cases for this addition?

**Review by ishantmehtaharness — APPROVED** [review]
looks good with 1 comment on testing part

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalChangeWindow.tsx (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalChangeWindow.tsx:114]: Coverage is already present for the error handling in it's parent component - but now i have added for this component as well.
Thanks!

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by namanharness — APPROVED** [review]

**Review by litvinovstas — APPROVED** [review]

**Pull request merged**